src/dotnet/CarbonAware.Tools.WattTimeClient/WattTimeClient.cs

```diff
@@ -0,0 +1,327 @@
+﻿using Microsoft.Extensions.Options;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using CarbonAware.Tools.WattTimeClient.Model;
+using System.Web;
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+using System.Security.Authentication;
+using System.Net.Mime;
+using System.Runtime.CompilerServices;
+using System.Net;
+using CarbonAware.Tools.WattTimeClient.Configuration;
+using CarbonAware.Tools.WattTimeClient.Constants;
+
+namespace CarbonAware.Tools.WattTimeClient;
+
+public class WattTimeClient : IWattTimeClient
+{
+    private const string BaseUrl = "https://api2.watttime.org/v2/";
```

Comment:
>private const string BaseUrl = "https://api2.watttime.org/v2/"; [](http://example.com/codeflow?start=4&length=63)

suggestion: might we want this configured outside the app, so we can point to a test-endpoint when needed, either watttime's or ours (aka: json-server: [https://www.npmjs.com/package/json-server](https://www.npmjs.com/package/json-server))

Reply:
Good call. fixed